src/audio/copier/copier.c

```diff
@@ -457,6 +477,72 @@ static int create_dai(struct comp_dev *parent_dev, struct copier_data *cd,
 	return 0;
 }
 
+#if CONFIG_COMP_IPCGTW
+static int create_ipcgtw(struct comp_dev *parent_dev, struct copier_data *cd,
+			 struct comp_ipc_config *config,
+			 const struct ipc4_copier_module_cfg *copier)
+{
+	const struct comp_driver *drv;
+	const struct sof_uuid uuid = {0xa814a1ca, 0x0b83, 0x466c, {0x95, 0x87, 0x2f,
+				      0x35, 0xff, 0x8d, 0x12, 0xe8}};
+	struct comp_dev *dev;
+
+	drv = ipc4_get_drv((uint8_t *)&uuid);
+	if (!drv)
+		return -EINVAL;
+
+	/* create_endpoint_buffer() uses this value to choose between input and
+	 * output formats in copier config to setup buffer. For this purpose
+	 * IPC gateway should be handled similarly as host gateway.
+	 */
+	config->type = SOF_COMP_HOST;
+
+	int ret = create_endpoint_buffer(parent_dev, cd, config, copier, ipc4_gtw_none, false, 0);
+
+	if (ret < 0)
```

Comment:
It would be good to have some error message here for ease of debug

Reply:
create_endpoint_buffer() (and buffer_alloc() inside it) do log in case of error. So it is not very important to also log an error here. The code was copy-pasted from create_host(), to fix error log it would be better to fix it also in other places where create_endpoint_buffer() is called: create_host() and init_dai(). So I left it as is, hope it's OK.